lstchain/reco/utils.py

```diff
@@ -55,6 +59,89 @@
 
 
 log = logging.getLogger(__name__)
+def rotate(flat_object, degree=0, origin=(0, 0)):
+    """
+    Rotate 2D object around given axle
+
+    Parameters:
+    -----------
+    array-like flat_object: 2D object to rotate
+    tuple origin: rotation axle coordinates
+    int degree: rotation angle in degrees
+
+    Returns:
+    --------
+    NDArray with new coordinates
+    """
+    angle = np.deg2rad(degree)
+    rotation_matrix = np.array([[np.cos(angle), -np.sin(angle)],
+                                [np.sin(angle), np.cos(angle)]])
+    rotation_axis_coordinates = np.asarray(origin)
+    res = [(rotation_matrix @ (point.T-rotation_axis_coordinates.T) +
+                       rotation_axis_coordinates.T).T for point in np.atleast_2d(flat_object)]
+    return res
+
+
+def extract_source_position(data, observed_source_name):
+    """
+    Extract source position from data
+
+    Parameters:
+    -----------
+    pandas.DataFrame data: input data
+    str observed_source_name: Name of the observed source
+
+    Returns:
+    --------
+    2D array of coordinates of the source in form [(x),(y)]
+    """
+    observed_source = SkyCoord.from_name(observed_source_name)
+    obstime = pd.to_datetime(data['dragon_time'], unit='s')
+    pointing_alt = u.Quantity(data['alt_tel'], u.rad, copy=False)
+    pointing_az = u.Quantity(data['az_tel'], u.rad, copy=False)
+    # FIXME hardcoded focal length next line
+    source_pos_camera = radec_to_camera(observed_source, obstime,
+                                        pointing_alt, pointing_az, focal=28*u.m)
+    source_position = [source_pos_camera.x.to_value(), source_pos_camera.y.to_value()]
+    return source_position
+
+
+def compute_theta2(data, source_position, conversion_factor=2.0):
+    """
+    Computes a square of theta (angle from z-axis) from camrea frame coordinates
+
+    Parameters:
+    -----------
+    pandas.DataFrame data: Input data
+    2D array (x,y) source_position: Observed source name
+    float conversion_factor: Conversion factor (default 0.1/0.05 deg/m)
+
+    Returns:
+    -------
+    Array with `theta2` values
+    """
+    reco_src_x = data['reco_src_x']
+    reco_src_y = data['reco_src_y']
+    return conversion_factor**2 * ((source_position[0]-reco_src_x)**2 + (source_position[1]-reco_src_y)**2)
+
+
+def compute_alpha(data):
+    """
+    Computes the angle between the shower major axis and polar angle of the shower centroid
+
+    Parameters:
+    -----------
+    pandas.DataFrame data: Input data
+
+    Returns:
+    --------
+    Array with `alpha` values
+    """
+    # phi and psi range [-np.pi, +np.pi]
+    alpha = np.mod(data['phi'] - data['psi'], np.pi)  # alpha in [0, np.pi]
+    alpha = np.minimum(np.pi - alpha, alpha)  # put alpha in [0, np.pi/2]
+
+    return np.rad2deg(alpha)
```

Comment:
Interestingly, alpha was already computed for the source-dep analysis but with a [different implementation](https://github.com/cta-observatory/cta-lstchain/blob/4a903e03ea1a229065fb53adb7faec2a269db20e/lstchain/reco/dl1_to_dl2.py#L481):
`src_dep_params['alpha'] = np.rad2deg(np.arctan(np.abs(miss / disp)))`

Could you get this straight with @SeiyaNozaki, propose a single implementation and use it at both places?